activesupport/lib/active_support/encrypted_configuration.rb

```diff
@@ -0,0 +1,54 @@
+require "yaml"
+require "active_support/encrypted_file"
+require "active_support/core_ext/object/inclusion"
+
+module ActiveSupport
+  class EncryptedConfiguration < EncryptedFile
+    delegate :dig, :fetch, :[], :[]=, to: :config
+
+    def initialize(config_path:, key_path:, env_key:, serializer: :yaml)
+      super content_path: config_path, key_path: key_path, env_key: env_key
+      @serializer = validated_serializer(serializer)
+    end
+
+    # Allow a config to be started without a file present
+    def read
+      super
+    rescue ActiveSupport::EncryptedFile::MissingContentError
+      ""
+    end
+
+    def config
+      @config ||= deserialize(read).deep_symbolize_keys
```

Comment:
Should we use `ActiveSupport::InheritedOptions` here on top of this? It would allow things like `Rails.credentials.aws_access_key`?